# Теория вероятностей и математическая статистика (семинары)
## Урок 5. Тестирование гипотез

## Задача 4
Есть ли статистически значимые различия в росте дочерей?

Рост матерей: 172, 177, 158, 170, 178,175, 164, 160, 169, 165
    
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [1]:
import numpy as np
from scipy import stats

In [2]:
x = np.array([172, 177, 158, 170, 178,175, 164, 160, 169, 165])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])

Рост матерей и рост взрослых дочерей должны быть зависимыми величинами, но нам даны выборки разной длины. Полагаем, что это какая то ошибка, и выбросим одно значение из выборки X для выравнивания.

In [3]:
from random import randint
x = np.delete(x,randint(0,len(x)))
x

array([172, 158, 170, 178, 175, 164, 160, 169, 165])

Вычисляем статистику

In [8]:
stats.ttest_rel(y,x)

Ttest_relResult(statistic=0.04510640748198446, pvalue=0.9651280953848651)

Найдем критическую точку по таблице t критерия Стьюдента

In [7]:
v = 2.101
v

2.101

**Вывод** - Вычисленная нами статистика (0.045) меньше, чем найденная по таблице критерия Стьюдента (2.101).

На основании этого - гипотеза принимается.